src/main/java/com/vividsolutions/jcs/conflate/coverage/Coverage.java

```diff
@@ -121,10 +121,10 @@ public void computeAdjustedFeatureUpdates(double distanceTolerance, boolean inte
                 // polygon into a MultiPolygon (only if it is noded)
                 if (!g.isValid()) g = g.buffer(0);
                 // don't update geometry if it's not valid
-                Debug.println("    adjusted geometry for : " + cgf.getFeature().getID() + " " +
-                    ((g==null)?"g=null   ":"g!=null   ") + (g.isValid()?"g.isValid()":"!g.isValid()"));
-                Feature originalFeat = cgf.getFeature();
-                Feature f = originalFeat.clone(false);
+//                Debug.println("    adjusted geometry for : " + cgf.getFeature().getID() + " " +
+//                    ((g==null)?"g=null   ":"g!=null   ") + (g.isValid()?"g.isValid()":"!g.isValid()"));
```

Comment:
This particular statement is doubly as bad since `g.isValid()` is evaluated again!

Reply:
these debug pritns should use  `com.vividsolutions.jump.workbench.Logger`

there is also a `Logger.isDebugEnabled()` to build them conditionally only.

there is a trace support with limiting to packages/classes planned for the futurem, but until then this should suffice.

Reply:
Java builds the string regardless of whether the Logger debug is enabled.

Reply:
i'm good with commenting. but you can as well (pseudo code)

```java
if (Logger.isDebugEnabled())
  Logger.debug("adjusted geometry for : " + cgf.getFeature().getID() + " " +((g==null)?"g=null   ":"g!=null   ") + (g.isValid()?"g.isValid()":"!g.isValid()"));
```

but sure,  `g.isValid()` should be limited to one call anyway.